bin/import-test-content.php

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/php
+<?php
+
+namespace WPOrg_Learn\Bin\ImportTestContent;
+
+/**
+ * CLI script for generating local test content, fetched from the live learn.wordpress.org site.
+ *
+ * This needs to be run in a wp-env, for example:
+ *
+ * yarn run wp-env run cli "php bin/import-test-content.php"
+ */
+
+// This script should only be called in a CLI environment.
+if ( 'cli' != php_sapi_name() ) {
+    die();
+}
+
+
+$opts = getopt( '', array( 'post:', 'url:', 'abspath:', 'age:' ) );
+
+require dirname( dirname( __FILE__ ) ) . '/wp-load.php';
+
+// Refuse to run on a production site.
+$run_on_hosts = [
+	'localhost',
+	'127.0.0.1',
+];
+if ( !in_array( parse_url( get_site_url(), PHP_URL_HOST ), $run_on_hosts ) ) {
+	die( "Not safe to run on " . get_site_url() );
+}
+
+function sanitize_meta_input( $meta ) {
+	$meta = (array( $meta ) );
+	foreach ( $meta as $k => $v ) {
+		if ( is_array( $v ) ) {
+			$meta[ $k ] = implode( ',', $v );
+		}
+	}
+
+	return $meta;
+}
+
+function import_rest_to_posts( $rest_url ) {
+	$response = wp_remote_get( $rest_url );
+	if ( is_wp_error( $response ) ) {
+		die( $response->get_error_message() );
+	}
+
+	$body = wp_remote_retrieve_body( $response );
+	$data = json_decode( $body );
+
+	foreach ( $data as $post ) {
+		echo "Got {$post->type} {$post->id} {$post->slug}\n";
+
+
+		// Surely there's a neater way to do this.
```

Comment:
Yeah you're probably right. It's not performance that worried me, it's having to explicitly transform fields in the json to differently-named fields in the php array. That's not necessarily a bad thing though since it gives us control over exactly what is and is not inserted.

Reply:
Ah, I see. Yeah, some clever mapping thing would probably make the code more complicated 👍🏻